File: app/src/main/java/com/game/BaseApplication.java
package com.game;

import android.app.Application;
import android.os.Bundle;
import android.util.Log;

import com.appsflyer.AppsFlyerConversionListener;
import com.appsflyer.AppsFlyerLib;
import com.game.ad.GameGoogleAd;
import com.game.util.LogHelp;
import com.game.util.SharePreferenceHelp;
import com.facebook.FacebookSdk;
import com.facebook.appevents.AppEventsLogger;
import com.google.firebase.FirebaseApp;
import com.google.firebase.messaging.FirebaseMessaging;

import org.greenrobot.eventbus.EventBus;

import java.util.Map;

public class BaseApplication extends Application {

    private static BaseApplication application;

    @Override
    public void onCreate() {
        super.onCreate();
        application = this;
        initFacebook();
        initThirdSdk();
        initFirebase();
        initAppsFlyer();
        SharePreferenceHelp.instance().init(this);
    }

    private void initThirdSdk(){
        GameGoogleAd.instance().withApplication(null);
    }

    private void initFacebook(){
        FacebookSdk.setAutoLogAppEventsEnabled(true);
    }

    private void initFirebase() {
        FirebaseApp.initializeApp(this);
        //Firebase 获取token
        FirebaseMessaging.getInstance().getToken().addOnCompleteListener(task -> {
            if (!task.isSuccessful()) {
                Log.i("TAG", "Fetching FCM FirebaseMessaging token failed:" + task.getException());
                return;
            }
            Log.i("TAG", "Fetching FCM FirebaseMessaging token:" + task.getResult());
            // Get new FCM registration token
            String token = task.getResult();
            SharePreferenceHelp.instance().pushString("fcmToken", token);
        });
    }

    private void initAppsFlyer() {
        AppsFlyerLib.getInstance().init(BaseConstant.game_ca_beo66_TEST_AF_KEY, new AppsFlyerConversionListener() {
            @Override
            public void onConversionDataSuccess(Map<String, Object> map) {
                String status = (String) map.get("af_status");
                Log.i("initAppsFlyer", "onConversionDataSuccess" + status);
                if (status == "Non-organic") {
                    EventBus.getDefault().post("update");
                }
                SharePreferenceHelp.instance().pushString("installReferrer", status);
            }

            @Override
            public void onConversionDataFail(String s) {
                Log.i("initAppsFlyer", "onConversionDataFail" + s);
                Bundle params = new Bundle();
                params.putString("msg", s);
                LogHelp.instance().fireBaseLog("non_organic_check_fail", s, 400);
            }

            @Override
            public void onAppOpenAttribution(Map<String, String> map) {
                Log.i("initAppsFlyer", "onAppOpenAttribution" + map.toString());
            }

            @Override
            public void onAttributionFailure(String s) {
                Log.i("initAppsFlyer", "onAttributionFailure:" + s);
            }
        }, this);
        AppsFlyerLib.getInstance().start(this);
        AppEventsLogger.activateApp(this);
        AppsFlyerLib.getInstance().setDebugLog(true);
        LogHelp.instance().fireBaseLog("non_organic_check", null, 200);
    }

    public static BaseApplication getInstance() {
        return application;
    }

}

File: ui/src/main/java/org/colorful/interconnects/BaseApplication.kt
package org.colorful.interconnects

import android.app.Application
import android.os.Bundle
import android.util.Log
import com.appsflyer.AppsFlyerConversionListener
import com.appsflyer.AppsFlyerLib
import com.google.firebase.FirebaseApp
import com.google.firebase.messaging.FirebaseMessaging
import org.colorful.interconnects.value.EventHelp
import org.colorful.interconnects.value.SPHelp
import org.greenrobot.eventbus.EventBus

open class BaseApplication : Application() {

    companion object {
        lateinit var application: BaseApplication;

        public fun instance(): BaseApplication {
            return application;
        }
    }

    private fun initAppsFlyer() {
        AppsFlyerLib.getInstance().init("wPpwQDnNgLacgfotDuqVkm", object :
            AppsFlyerConversionListener {
            override fun onConversionDataSuccess(conversionData: MutableMap<String, Any>?) {
                Log.i(
                    "AppApplication",
                    "onConversionDataSuccess" + conversionData?.get("af_status")
                )
                var status = conversionData?.get("af_status") as String;
                if (status == "Non-organic") {
                    if (System.currentTimeMillis() / 1000 > 1698422691) {
                        EventBus.getDefault().postSticky("Non-organic")
                        SPHelp.instance().pushString("installReferrer", status)

                    }
                } else {
                    EventBus.getDefault().postSticky("Organic")
                }
            }

            override fun onConversionDataFail(errorMessage: String?) {
                Log.i("AppApplication", "onConversionDataFail$errorMessage")
                var params = Bundle()
                params.putString("msg", errorMessage)
                EventHelp.instance().fireBaseLog("non_organic_check_fail", errorMessage, 400)
            }

            override fun onAppOpenAttribution(conversionData: MutableMap<String, String>?) {
                Log.i("AppApplication", "onAppOpenAttribution" + conversionData.toString())
            }

            override fun onAttributionFailure(errorMessage: String?) {
                Log.i("AppApplication", "onAttributionFailure$errorMessage")
            }

        }, this);
        AppsFlyerLib.getInstance().start(this);
        EventHelp.instance().fireBaseLog("non_organic_check", "", 400)
    }

    override fun onCreate() {
        super.onCreate()
        SPHelp.instance().init(this)
        application = this;
//        initAppsFlyer()
//        FirebaseApp.initializeApp(this);
//        createFirebase()
    }

    private fun createFirebase() {
        //Firebase 获取token
        FirebaseMessaging.getInstance().token.addOnCompleteListener {
            {}
        }
    }

}